### ISSUE TITLE
Dosdebug *mhp_getreg() mix return values with error states

### ISSUE THREAD

**Issue by andrewbird** [stsp/dosemu2#616]
I notice that the functions `dpmi_mhp_getreg()` and `mhp_getreg()` return error states `0: not in DPMI` and `-1: match not found`. Both of these may be actual values held in a register being queried so how should a caller distinguish?

**Comment by stsp** [comment]
Maybe by the use of assert() instead of returning
any error?
PS: vacations were much nicer than coding. :) Sigh.

**Comment by andrewbird** [comment]
I was thinking of changing 
```C
-static unsigned long mhp_getreg(char * regn)
+static int mhp_getreg(char *regn, unsigned int *val, int *typ)
```
and have it return the error, and set the result via the val pointer etc / type etc

Ahh good to know about the hols, I had just asked in the PR you just closed.

**Comment by stsp** [comment]
But do you really need to return an error?
If this is an error that should never happen,
then why to bother.

**Comment by andrewbird** [comment]
i'm not too confident about my understanding of DPMI / protected mode stuff, so if you're happy asserting then I'll fix it that way.

**Comment by stsp** [comment]
Its not about DPMI, its just that dosdebug
should never call `dpmi_mhp_getreg()` in
v86 (non-PM) mode. If this is so, things can
be changed to assert.

**Comment by andrewbird** [comment]
I started to code this up, but I think `mhp_getreg()` might be called by a dosdebug user with a non existent register name. I don't think asserting is the right thing to do here. e.g.
```C
static unsigned long mhp_getreg(char *regn)                                     
{                                                                               
  int symreg = decode_symreg(regn);                                             
                                                                                
  assert(symreg != -1);                                                         
                                                                                
  if (IN_DPMI)                                                                  
    return dpmi_mhp_getreg(symreg);                                             
                                                                                
  switch (symreg) {                                                             
    case _SSr: return SREG(ss);                                                 
    case _CSr: return SREG(cs);                                                 
    case _DSr: return SREG(ds);                                                 
```

**Comment by andrewbird** [comment]
I think I might need to change `mhp_getreg()` to take a symreg value, and decode the string & validate before its call. I'll have a play...

**Comment by andrewbird** [comment]
Have a look at https://github.com/andrewbird/dosemu2/commit/4620814a181560d10bd513b39cc69fb4eb0236a6 and see if it's okay style wise? I didn't test it yet, but I'd to think it's along the right lines before I spend too much effort on it.

**Comment by stsp** [comment]
I think its fine as long as the initial code
was also an utter mess. You haven't untangled
it to some reasonable margin, but lets say
that any improvement is better than nothing. :)

**Comment by andrewbird** [comment]
Forgot to add fixes #616 into the commit message!
